### PR TITLE
chore: enable exhaustruct linter

### PR DESCRIPTION
**Pull request by johnstcn**
Just enabling on `httpmw\.\w+` for now. We can add coverage as we go.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -527,6 +527,7 @@ distro() {
 
 	if [ -f /etc/os-release ]; then
 		(
+			# shellcheck disable=SC1091
```
Comment (johnstcn) [inline install.sh:530]: review: this was making lint fail on darwin

File: coderd/httpmw/realip.go [hunk 2]
```diff
@@ -56,7 +56,10 @@ func ExtractRealIP(config *RealIPConfig) func(next http.Handler) http.Handler {
 // configuration and headers. It does not mutate the original request.
 func ExtractRealIPAddress(config *RealIPConfig, req *http.Request) (net.IP, error) {
 	if config == nil {
-		config = &RealIPConfig{}
+		config = &RealIPConfig{
+			TrustedOrigins: []*net.IPNet{},
+			TrustedHeaders: []string{},
```
Comment (mafredri) [inline coderd/httpmw/realip.go:61]: To match previous input, these would be `nil`. I doubt there's a meaningful difference, but something to keep in mind.

**Review by mafredri — APPROVED** [review]
Looks great, I think this will can be a really valuable tool which forces us to recognize edge-cases and expanded API surface in relevant places!

File: coderd/coderd.go [hunk 3]
```diff
@@ -390,6 +390,7 @@ func New(options *Options) *API {
 		RedirectToLogin:             false,
 		DisableSessionExpiryRefresh: options.DeploymentValues.DisableSessionExpiryRefresh.Value(),
 		Optional:                    false,
+		SessionTokenFunc:            nil, // Default behavior
```
Comment (Emyrk) [inline coderd/coderd.go:393]: Do we need to have the `// Default behavior` comment?

The comment on the field says: `If nil, the default behavior is used.`

**Review by Emyrk — APPROVED** [review]
:+1: 

One thing that we can do if some structs are just annoying, it maybe adopt a `New()` style init and use chaining to fill fields. If there are big structs that annoying to populate each time.

File: coderd/coderd.go (hunk 3 above)
Comment (johnstcn) [reply coderd/coderd.go:393]: We don't need the comment, I was just being super explicit about what it means.

**Pull request merged**